src/dmd/aliasthis.d

```diff
@@ -72,69 +71,6 @@ extern (C++) final class AliasThis : Dsymbol
     }
 }
 
-Expression resolveAliasThis(Scope* sc, Expression e, bool gag = false)
-{
-    for (AggregateDeclaration ad = isAggregate(e.type); ad;)
-    {
-        if (ad.aliasthis)
-        {
-            uint olderrors = gag ? global.startGagging() : 0;
-            Loc loc = e.loc;
-            Type tthis = (e.op == TOK.type ? e.type : null);
-            e = new DotIdExp(loc, e, ad.aliasthis.ident);
-            e = e.expressionSemantic(sc);
-            if (tthis && ad.aliasthis.sym.needThis())
-            {
-                if (e.op == TOK.variable)
-                {
-                    if (auto fd = (cast(VarExp)e).var.isFuncDeclaration())
-                    {
-                        // https://issues.dlang.org/show_bug.cgi?id=13009
-                        // Support better match for the overloaded alias this.
-                        bool hasOverloads;
-                        if (auto f = fd.overloadModMatch(loc, tthis, hasOverloads))
-                        {
-                            if (!hasOverloads)
-                                fd = f;     // use exact match
-                            e = new VarExp(loc, fd, hasOverloads);
-                            e.type = f.type;
-                            e = new CallExp(loc, e);
-                            goto L1;
-                        }
-                    }
-                }
-                /* non-@property function is not called inside typeof(),
-                 * so resolve it ahead.
-                 */
-                {
-                    int save = sc.intypeof;
-                    sc.intypeof = 1; // bypass "need this" error check
-                    e = resolveProperties(sc, e);
-                    sc.intypeof = save;
-                }
-            L1:
-                e = new TypeExp(loc, new TypeTypeof(loc, e));
-                e = e.expressionSemantic(sc);
-            }
-            e = resolveProperties(sc, e);
-            if (!gag)
-                ad.aliasthis.checkDeprecatedAliasThis(loc, sc);
-            else if (global.endGagging(olderrors))
-                e = null;
-        }
-
-        import dmd.dclass : ClassDeclaration;
-        auto cd = ad.isClassDeclaration();
-        if ((!e || !ad.aliasthis) && cd && cd.baseClass && cd.baseClass != ClassDeclaration.object)
-        {
-            ad = cd.baseClass;
-            continue;
-        }
-        break;
-    }
-    return e;
-}
-
```

Comment:
The function below `checkDeprecatedAliasThis` should also be moved wherever `resolveAliasThis` goes and also it should be made `private`. There are 2 reasons for this: firstly, because it is a helper function and only `resolveAliasThis` calls it, secondly, because it also imports dsymbolsem and therefore has a dependency to a module that does semantic analysis.